### PR TITLE
Savestates Fixes (part 2 out of 3)

### PR DESCRIPTION
**Pull request by elad335**
Savestates should work now fully using ASMJIT.

### CODE REVIEW

**Comment by Ordinary205** [comment]
RPCS3 closes after linking PPU modules.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12130562/RPCS3.log.gz)

**Comment by Ordinary205** [comment]
I do confirm that RPCS3 no longer exits when building PPU modules, good work elad.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12130928/RPCS3.log.gz)

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -2369,7 +2373,24 @@ void main_window::CreateConnects()
 		fs::stat_t raw_stat{};
 		fs::stat_t archived_stat{};
 
-		if ((!fs::stat(raw_file_path, raw_stat) || raw_stat.is_directory) || (!fs::stat(archived_path, archived_stat) || archived_stat.is_directory) || (raw_stat.size == 0 && archived_stat.size == 0))
+		auto get_stat_for_existence_check = [](const std::string& filename, fs::stat_t& stat)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:2376]: This should be a different PR

**Comment by elad335** [comment]
Please look for perrformance regressions with pr.

**Comment by Ordinary205** [comment]
Heres a performance results that I've tested on RDR:

Master: 26.8/28.4/29.5 FPS
RDR FPS TEST 1 (2)

PR: 25/26.6/27.5 FPS
RDR FPS TEST 2 (2)
